test/DelimitedDataParser.Test/ParserTest.Reader.cs

```diff
@@ -1892,5 +1893,87 @@ public void ParseReader_HasRows_ReturnsTrueForHeaderAndDataRows()
 
             Assert.True(reader.HasRows);
         }
+
+        [Fact]
+        public void Supports_Varying_Column_Counts_In_Enumerator_With_Column_Headers()
+        {
+            var content = @"Field 1,Field 2,Field 3" + Environment.NewLine
+                + @"Data 1,Data 2" + Environment.NewLine
+                + @"Data 5" + Environment.NewLine
+                + @"Data 6,Data 7,Data 8, data 9";
+
+            using (var stringReader = new StringReader(content))
```

Comment:
This test appears to be inconsistent with existing tests in the way that it is parsing the test input. Let's modify this test to make use of `GetTextReader()`, rather working with a `StringReader` directly.